src/marray.ts

```diff
@@ -1,7 +1,12 @@
 import * as ts from './tree-structure';
 import * as tu from './tree-utils';
 
+let isNumber = (thing: number | string): thing is number => {
+  return !isNaN(Number.parseInt(thing.toString()));
```

Comment:
Seems like this will also convert "1.1" to the property 1 and will do parsing and conversion for every access with a number.

Something like

```js
function toInteger(key) { 
  if (Number.isInteger(key)) return key; 
  let ks = key.toString(); if (ks === '') return; let n = Number(ks); if (Number.isInteger(n)) return n; return null; 
}
```

will make sure that we have good performance for the standard case (numbers)

```
> console.time(); for (let k = 0; k < 1000000; ++k) toInteger(k); console.timeEnd()
default: 14.652ms
```

while still doing OK if strings are passed and behaving like an array:

```
> console.time(); for (let k = 0; k < 1000000; ++k) toInteger(k.toString()); console.timeEnd()
default: 173.799ms
```

(we can use `let maybeInt = toInteger(key); if (maybeInt != null) ...` to check if it was successfully converted to integer)